File: problems/0069-sqrtx/solution.py
class Solution(object):
    def mySqrt(self, x):
       
       if x==1:
        return x
       elif x==0:
        return x  

       else: 
        start=1
        end=x
        while(start<=end):
            mid=start+(end-start)//2
            if((mid*mid)==x):
                return int(mid)
            elif(x<(mid*mid)):
                end=mid-1
            else:
                start=mid+1

        return end

File: problems/0069-sqrtx/solution.c
int mySqrt(int x) {
    long int start=1,end=x,mid;
    if(x==0 || x==1)
       return x;
    else{
        while(start<=end){
            mid=start+(end-start)/2;
            if(mid*mid==x)
               return mid;
            else if(x<mid*mid)
               end=mid-1;
            else
                start=mid+1;  

        }
        return end;
    }
          
}
